### PR TITLE
Perf/issue 520/update wcp module

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**

### CODE REVIEW

**Review by delehef — APPROVED** [review]

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java [hunk 1]
```diff
@@ -24,93 +25,104 @@
...
+  private boolean bit1;
+  private Boolean bit2;
+  private final Boolean bit3;
+  private final Boolean bit4;
   private Boolean resLo;
 
   final List<Boolean> bits = new ArrayList<>(16);
 
-  public WcpOperation(OpCode opCode, Bytes32 arg1, Bytes32 arg2) {
+  public WcpOperation(byte opCode, Bytes32 arg1, Bytes32 arg2) {
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java:73]: Maybe `inst` or `wcpInst` would be better than byte ?

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java [hunk 2]
```diff
@@ -201,7 +212,42 @@ void trace(Trace trace, int stamp) {
...
+          return 0;
+        } else {
+          final ArrayList<Integer> sizes = new ArrayList<>(6);
+          sizes.add(this.arg1.slice(0, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg2.slice(0, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg1.slice(LLARGE, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg2.slice(LLARGE, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.adjHi.trimLeadingZeros().size());
+          sizes.add(this.adjLo.trimLeadingZeros().size());
+          return Collections.max(sizes) - 1;
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java:247]: Is it ever possible that max(sizes) returns 0 E.g. what happens if all arguments to the instruction are 0 ?

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java:247]: Oh my bad, you consider that case above :slightly_smiling_face:

**Review by OlivierBBB — APPROVED** [review]
LGTM

**Pull request merged**